affinity/cpp-20/d0796r2.md

```diff
@@ -170,150 +187,134 @@ This feature could be easily scaled to heterogeneous and distributed systems, as
+
 ### Querying relative affinity
 
-The `affinity_query` class template provides an abstraction for a relative affinity value between two `execution_resource`s, derived from a particular `affinity_operation` and `affinity_metric`. The `affinity_query` is templated by `affinity_operation` and `affinity_metric` and is constructed from two `execution_resource`s. An `affinity_query` does not mean much on it's own, instead a relative magnitude of affinity can be queried by using comparison operators. If nessesary the value of an `affinity_query` can also be queried through `native_affinity`, though the return value of this is implementation defined.
+The `affinity_query` class template provides an abstraction for a relative affinity value between an execution resource and a memory resource, derived from a particular `affinity_operation` and `affinity_metric`. The `affinity_query` is templated by `affinity_operation` and `affinity_metric` and is constructed from an execution resource and memory resource. An `affinity_query` does not mean much on it's own, instead a relative magnitude of affinity can be queried by using comparison operators. If nessesary the value of an `affinity_query` can also be queried through `native_affinity`, though the return value of this is implementation defined.
 
-Below *(listing 3)* is an example of how you can query the relative affinity between two `execution_resource`s.
+Below *(listing 3)* is an example of how you can query the relative affinity between two execution resources.
 
 ```cpp
-auto systemLevelResources = execution::this_system::resources();
-auto memberResources = systemLevelResources.resources();
+auto exec0 = *execution::this_system::execution_resource().begin();
+auto exec1 = *++execution::this_system::execution_resource().begin();
+auto mem0  = *exec0.memory_resource().begin();
 
-auto relativeLatency01 = execution::affinity_query<execution::affinity_operation::read,
-  execution::affinity_metric::latency>(memberResources[0], memberResources[1]);
+auto relativeLatency0 = execution::affinity_query<execution::affinity_operation::read,
+  execution::affinity_metric::latency>(exec0,mem0);
 
-auto relativeLatency02 = execution::affinity_query<execution::affinity_operation::read,
-  execution::affinity_metric::latency>(memberResources[0], memberResources[2]);
+auto relativeLatency1 = execution::affinity_query<execution::affinity_operation::read,
+  execution::affinity_metric::la1Gtency>(exec1,mem0);
 
-auto relativeLatency = relativeLatency01 > relativeLatency02;
+auto relativeLatency = relativeLatency0 > relativeLatency1;
 ```
 *Listing 3: Example of querying affinity between two `execution_resource`s.*
 
 > [*Note:* This interface for querying relative affinity is a very low-level interface designed to be abstracted by libraries and later affinity policies. *--end note*]
 
-### Execution context
```

Comment:
Seems you are removing entirely the Execution context (I thought it was just moved elsewhere at first). Is that intentional?
The text refers to the concept of Execution context, so I think we should keep it well defined, even if we make changes to the interface itself later.

Reply:
The discussion of execution context is still in the paper, just no immediate proposal for a concrete execution context such as static or dynamic thread pools.